app/screens/home/channel_list/categories_list/categories/categories.tsx

```diff
@@ -49,12 +49,19 @@ const Categories = ({
     const isTablet = useIsTablet();
     const switchingTeam = useTeamSwitch();
     const teamId = categories[0]?.teamId;
+    const showAllCategory = onlyUnreads && !unreadsOnTop;
+
     const categoriesToShow = useMemo(() => {
+        const orderedCategories = [...categories];
+        orderedCategories.sort((a, b) => a.sortOrder - b.sortOrder);
+
+        if (onlyUnreads && unreadsOnTop) {
+            return orderedCategories;
+        }
```

Comment:
Still not sure about this change.

If `unreadsOnTop` is set, it should take preference over `onlyUnreads`. So... we want the `Unreads` category and the rest of the categories, right?

So... if both are true, the next if (`onlyUnreads && !unreadsOnTop`) will skip, and the `unreadsOnTop` if will do the right thing (show unreads and the rest of categories).

Therefore... There is no need to any changes on this `useMemo`. Just the changes you have later in the render, right?

Reply:
@larkox  If `unreadsOnTop` is true, then it is rendering `<EmptyState>` with all the categories, because  `onlyUnreads ` is also true, that's why  `return orderedCategories;`